core/services/offchainreporting/delegate.go

```diff
@@ -143,24 +146,13 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err
 	)
 	services = append(services, tracker)
 
-	var peerID p2pkey.PeerID
-	if concreteSpec.P2PPeerID != nil {
-		peerID = *concreteSpec.P2PPeerID
-	} else {
-		k, err2 := d.keyStore.P2P().GetOrFirst(chain.Config().P2PPeerID().Raw())
```

Comment:
I see this GetOrFirst call removed from here, but I can't find where it moved to

Reply:
Good spot, this turned out to be a huge iceberg which revealed some bugs in P2PPeerID handling. Working on it.